src/secrets/CommandCat.ts

```diff
@@ -49,16 +51,55 @@ class CommandGet extends CommandPolykey {
           },
           logger: this.logger.getChild(PolykeyClient.name),
         });
-        const response = await binUtils.retryAuthentication(
-          (auth) =>
-            pkClient.rpcClient.methods.vaultsSecretsGet({
-              metadata: auth,
-              nameOrId: secretPath[0],
-              secretName: secretPath[1],
-            }),
-          meta,
-        );
-        const secretContent = Buffer.from(response.secretContent, 'binary');
+        if (secretPaths.length === 0) {
+          await new Promise((resolve, reject) => {
+            let stdout: string = '';
+
+            const cleanup = (code?: any) => {
+              process.removeAllListeners('data');
+              process.removeAllListeners('exit');
+              process.removeAllListeners('error');
+              if (code == null) reject();
+              else reject(code);
+            };
+
+            process.stdin.on('data', (data) => {
+              stdout += data.toString();
+            });
+
+            process.stdin.once('error', (code) => cleanup(code));
+            process.stdin.once('close', () => cleanup());
+            process.stdin.once('end', () => {
+              process.stdout.write(stdout);
+              resolve(0);
+            });
+          });
+          return;
+        }
+
+        const response = await binUtils.retryAuthentication(async (auth) => {
+          const response = await pkClient.rpcClient.methods.vaultsSecretsGet();
+          await (async () => {
+            const writer = response.writable.getWriter();
+            let first = true;
+            for (const [vault, path] of secretPaths) {
+              await writer.write({
+                nameOrId: vault,
+                secretName: path,
+                metadata: first ? auth : undefined,
+              });
+              first = false;
+            }
+            writer.close();
+          })();
+          return response;
+        }, meta);
+
+        let rawSecretContent: string = '';
+        for await (const chunk of response.readable) {
+          rawSecretContent += chunk.secretContent;
+        }
+        const secretContent = Buffer.from(rawSecretContent, 'binary');
```

Comment:
So, I should open up a temporary file, then stream the file contents into that file to accumulate them, then after the incoming stream has concluded, I can just read the file contents and stream that to the output, saving memory for large files and concatenations.

Is that what you meant?

Reply:
If you were writing to a file then you'd open the file to get the fd and write to it in chunks rather than get all the contents in memory and do a single `writeFile`.

In this case we're writing to stdout so we just need to do `process.stdout.write()` each chunk as we get them. We never want to save the whole contents of a file in memory like this.

Actually, looking closer at the code, the format option doesn't make sense for this command, we shouldn't be using it or the output formatter. We should be writing the contents directly to stdout. Or just use the raw type on the output formatter. I don't think having a JSON format for CAT makes sense here.

Reply:
Following a later comment, I have ditched accumulating the file contents and using output formatters. I print out the contents to `stdout` directly.

On that note, I was using `process.stdout.end()` to indicate that we will not be sending any more data to the output stream, and closing it. 

However, after research, I have discovered that this practice is discouraged with standard out streams. They are managed by NodeJS, and will end when the process terminates. 

If we manually end the streams, then any attempts of writing to `stdout`, be it intentional or accidental, will raise an error which will propagate to the top, crashing the program.

As such, I have removed all usages of `process.stdout.end()`.

Reply:
Wouldn't a JSON format simply be applied on the outside? And in that case, we would look at using JSONL designed for streaming, not a JSON array.